JoinColony/colonyCDapp
# Claiming expenditure payout

## Description

This PR adds claiming expenditure payout functionality.

After finalization, you should see a Claim funds button appear that lets you claim all the payouts for which the claim delay has elapsed (see below for more info on how claim delay works in dev).

## Testing

1. Make sure you restart your entire dev environment as there was a network version upgrade.
2. Create and advance expenditures through all the stages to finalization.
3. Attempt to claim the funds. Note: It is expected that attempts to claim a locked token (e.g. from another colony) will fail.
4. If you set your wallet address as a recipient, you should see your balance increase by nearly the entire amount of all your payouts (less the network fee).

Scenarios I recommend testing:
- Multiple payouts
- Edited payouts (especially removing one of the payouts)
- Payouts with different claim delays

### Note on claim delay in dev

Time works differently in dev blockchain. This makes testing a bit tricky, since the UI uses the real time, whereas the dev blockchain time is "frozen" until the next block is mined. 

The consequence of this is that when you finalize an expenditure with non-zero claim delay, the UI should show the correct time at which you can claim the next payout, e.g.:
<img width="540" alt="image" src="https://github.com/JoinColony/colonyCDapp/assets/112586815/b5fd2f00-3c65-4d29-b3ee-46e8af692e7d">

But you won't be able to claim it until another block is mined and the blockchain time updates. To force mining a block, run the following command in truffle:

```js
web3.currentProvider.send({method: "evm_mine", params: []}, () => {})
```

You can also advance time by an arbitrary number of seconds:

```js
web3.currentProvider.send({method: "evm_increaseTime", params: [<seconds>]}, () => {})
// You need to mine a block after increasing time
web3.currentProvider.send({method: "evm_mine", params: []}, () => {})
```

Resolves #912
